ui/app/AppLayouts/Profile/popups/AddShowTokenModal.qml

```diff
@@ -0,0 +1,183 @@
+import QtQuick 2.13
+import QtQuick.Controls 2.13
+import QtQuick.Dialogs 1.3
+
+import utils 1.0
+
+import StatusQ.Controls 0.1
+
+import shared.popups 1.0
+import shared.controls 1.0
+import "../stores"
+
+// TODO: replace with StatusModal
+ModalPopup {
+    id: popup
+
+    property bool editable: true
+    property int marginBetweenInputs: 35
+    property string validationError: ""
+    property WalletStore walletStore
+
+    title: editable ?
+        //% "Add custom token"
+        qsTrId("add-custom-token")
+        : nameInput.text
+
+    height: editable ? 450 : 380
+
+    onOpened: {
+        addressInput.forceActiveFocus(Qt.MouseFocusReason)
+    }
+
+    function openEditable(){
+        addressInput.text = "";
+        nameInput.text = "";
+        symbolInput.text = "";
+        decimalsInput.text = "";
+        editable = true;
+        open();
+    }
+
+    function openWithData(address, name, symbol, decimals){
+        addressInput.text = address;
+        nameInput.text = name;
+        symbolInput.text = symbol;
+        decimalsInput.text = decimals;
+        editable = false;
+        open();
+    }
+
+
+    function validate() {
+        if (addressInput.text !== "" && !Utils.isAddress(addressInput.text)) {
+            //% "This needs to be a valid address"
+            validationError = qsTrId("this-needs-to-be-a-valid-address");
+        }
+        return validationError === ""
+    }
+
+    property var getTokenDetails: Backpressure.debounce(popup, 500, function (tokenAddress){
+        popup.walletStore.walletTokensModule.getTokenDetails(tokenAddress)
+    });
+
+    function onKeyReleased(){
+        validationError = "";
+        if (!validate() || addressInput.text === "") {
+            return;
+        }
+        Qt.callLater(getTokenDetails, addressInput.text)
+    }
+
+    Item {
+        Connections {
+            target: popup.walletStore.walletTokensModule
+            onTokenDetailsWereResolved: {
+                const jsonObj = JSON.parse(tokenDetails)
+                if (jsonObj.error) {
+                    validationError = jsonObj.error
+                    return
+                }
+                if (jsonObj.name === "" && jsonObj.symbol === "" && jsonObj.decimals === "") {
+                    //% "Invalid ERC20 address"
+                    validationError = qsTrId("invalid-erc20-address")
+                    return;
+                }
+
+                if (addressInput.text.toLowerCase() === jsonObj.address.toLowerCase()) {
```

Comment:
can this signal from him side not pass 5 params like
proc tokenDetailsWereResolved(string error, string name, string symbol, string decimals, string name)
This Json processing in qml feels like we are performing logic and UI should basically not do this as much as possible.

Reply:
I "only" copy pasted code here can we revisit the logic later?